src/coreclr/jit/importer.cpp

```diff
@@ -7883,6 +7883,18 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                                || (impStackTop().val->TypeGet() == TYP_BYREF)
 #endif
                         ;
+#ifdef TARGET_AMD64
+                    // If AVX512 is present and we are not checking for overflow, we do not need
+                    // a large node. In this case, we will not fallback to a helper function but
+                    // will use the intrinsic instead. This is done for all long/ulong to floating
+                    // point conversions. Hence setting the callNode to false to
+                    // avoid generating a large node.
+                    if (callNode && !ovfl && varTypeIsLong(impStackTop().val) &&
```

Comment:
I would keep this simple and omit this special case here. It is ok for the large node allocations to have false positives.